src/category_theory/structured_arrow.lean

```diff
@@ -208,4 +208,127 @@ def mk_id_terminal [full S] [faithful S] : is_terminal (mk (𝟙 (S.obj Y))) :=
 
 end costructured_arrow
 
+open opposite
+
+namespace structured_arrow
+
+/--
+For a functor `F : C ⥤ D` and an object `d : D`, we obtain a contravariant functor from the
+category of structured arrows `d ⟶ F.obj c` to the category of costructured arrows
+`F.op.obj c ⟶ (op d)`.
+-/
+def to_costructured_arrow (F : C ⥤ D) (d : D) :
```

Comment:
Originally, I included `@[simps]` attributes for this and the other three functors. Somehow, this seemed to have caused a timeout in `grothendieck.lean` (see [here](https://github.com/leanprover-community/mathlib/runs/3393408314)). I have no idea what's going on here, as I don't really know how to diagnose timeouts. For now, I removed the `simps` and just unfold the definitions manually when I need to. Any help is appreciated!